internal/core/textlayout/sharedparley.rs

```diff
@@ -1213,6 +1221,67 @@ pub fn draw_text(
     }
 }
 
+#[cfg(feature = "experimental-rich-text")]
+pub fn link_under_cursor(
+    scale_factor: ScaleFactor,
+    text: Pin<&dyn crate::item_rendering::RenderText>,
+    font_request: Option<FontRequest>,
+    size: LogicalSize,
+    cursor: PhysicalPoint,
+) -> Option<std::string::String> {
+    let str = text.text();
+
+    let layout_text = Text::RichText(parse_markdown(&str));
+
+    let (horizontal_align, vertical_align) = text.alignment();
+
+    let layout = layout(
+        layout_text,
+        scale_factor,
+        LayoutOptions {
+            horizontal_align,
+            vertical_align,
+            max_height: Some(size.height_length()),
+            max_width: Some(size.width_length()),
+            stroke: None,
+            font_request,
+            text_wrap: text.wrap(),
+            text_overflow: text.overflow(),
+            selection: None,
+            selection_foreground_color: None,
+            link_color: crate::Color::from_rgb_u8(64, 64, 255),
+        },
+    );
+
+    let Some(paragraph) = layout.paragraph_by_y(cursor.y_length()) else {
+        return None;
+    };
+
+    let cursor = parley::layout::cursor::Cursor::from_point(
```

Comment:
I just realised that my idea had one caveat: `from_point` will be fuzzy, so if a link is at the horizontal end of a line (let's say ltr) and you click to the right of that line, it will still activate. There's no precise hit-test yet I think, so your previous approach gave the correct behaviour, I think.

Related: https://github.com/linebender/parley/pull/447